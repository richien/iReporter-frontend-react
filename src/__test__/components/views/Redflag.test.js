import React from 'react';

import { shallow } from 'enzyme';
import Redlag, { cards } from '../../../components/views/incidents/Redflag';

function renderRedflag(args) {
  const defaultProps = {
    redFlags: [{}],
    errors: {}
  };
  const props = { ...defaultProps, ...args };
  return shallow(<Redlag {...props} />);
}

describe('Redlag', () => {
  it('should match snapshot', () => {
    const wrapper = renderRedflag();
    expect(wrapper).toMatchSnapshot();
  });

  it('should return cardItems given an array of redflag objects', () => {
    const redFlags = [
      {
        status: 'resolved',
        title: 'test',
        createdOn: 'Wed, 13 Feb 2019 00:00:00 GMT',
        id: '1'
      }
    ];

    const cardItems = cards(redFlags);
    expect('div').toEqual(cardItems[0].type);
  });
});
