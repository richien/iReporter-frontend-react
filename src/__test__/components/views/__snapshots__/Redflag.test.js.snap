// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redlag should match snapshot 1`] = `ShallowWrapper {}`;
